x-pack/plugins/enterprise_search/public/applications/workplace_search/views/content_sources/source_logic.ts

```diff
@@ -546,18 +566,18 @@ export const SourceLogic = kea<MakeLogicType<SourceValues, SourceActions>>({
       // Remove undefined values from params
       Object.keys(params).forEach((key) => params[key] === undefined && delete params[key]);
 
-      return http
-        .post(route, params)
-        .then(({ data }) => {
-          actions.setCustomSourceData(data);
-          successCallback();
-        })
-        .catch((e) => {
-          flashAPIErrors(e);
-          if (errorCallback) errorCallback();
-          throw new Error('Auth Error');
-        })
-        .finally(actions.setButtonNotLoading);
+      try {
+        const response = await HttpLogic.values.http.post(route, {
+          body: JSON.stringify({ params }),
+        });
+        actions.setCustomSourceData(response);
+        successCallback();
+      } catch (e) {
+        flashAPIErrors(e);
+        actions.setButtonNotLoading();
```

Comment:
This was in `finally` block before. Now it's only in `catch` block. Should we move it below the `try-catch` block?